lib/src/Uncertainty/Algorithm/Analytical/Analytical.cxx

```diff
@@ -42,18 +42,31 @@ Analytical::Analytical(const OptimizationAlgorithm & nearestPointAlgorithm,
 {
   if (physicalStartingPoint.getSize())
   {
-    LOGWARN("FORM physicalStartingPoint argument is deprecated");
+    LOGWARN("FORM/SORM physicalStartingPoint argument is deprecated");
     nearestPointAlgorithm_.setStartingPoint(physicalStartingPoint);
   }
 
   const UnsignedInteger dimension = event.getImplementation()->getFunction().getInputDimension();
-  if (nearestPointAlgorithm_.getStartingPoint().getDimension() != dimension)
-    throw InvalidArgumentException(HERE) << "Starting point dimension (" << nearestPointAlgorithm_.getStartingPoint().getDimension() << ") does not match event dimension (" << dimension << ").";
+
+  try 
+  {
+    if (nearestPointAlgorithm_.getStartingPoint().getDimension() != dimension)
+      throw InvalidArgumentException(HERE) << "Starting point dimension (" << nearestPointAlgorithm_.getStartingPoint().getDimension() << ") does not match event dimension (" << dimension << ").";
+  }
+  catch (const NotDefinedException &) // MultiStart algorithm
+  {
+    if (nearestPointAlgorithm_.getStartingSample()[0].getDimension() != dimension)
```

Comment:
you can write `getStartingSample().getDimension()` directly

Reply:
fixed